### ISSUE TITLE
[Accessibility]: Command palette doesn't restore focus after running commands

### ISSUE THREAD

**Issue by tjkandala** [sourcegraph/sourcegraph#36858]
### Audit type

Keyboard navigation

### User journey audit issue

#34061

### Problem description

The command palette can be summoned with the `ctrl` + `p` shortcut. When the user submits a command with `Enter` or closes the command palette with `esc`, the previously focused item isn't focused. Rather, focus is returned to the top of the page.

### Expected behavior

Previously focused element should receive focus once the command palette closes.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @muratsu @jjinnii @ryankscott - the "team/integrations" label was applied to this issue.
